sync/src/block/downloader/header.rs

```diff
@@ -133,9 +135,11 @@ impl HeaderDownloader {
             }
         } else {
             let pivot_header = self.pivot_header();
-            self.pivot = Pivot {
-                hash: pivot_header.parent_hash(),
-                total_score: self.pivot.total_score - pivot_header.score(),
+            if pivot_header.hash() != self.genesis_hash {
```

Comment:
Let's check parent header's information rather than checking its hash with an external parameter.
Genesis header always has a parent hash of zero.

Reply:
No, it's not.
https://github.com/CodeChain-io/codechain/blob/e1752be322cd0e84553ac522b9a33478ab2b5a23/core/res/cuckoo.json#L30 You can change the parent hash of the genesis block.

Reply:
Oh, I didn't know that.
Then how about checking its block number?